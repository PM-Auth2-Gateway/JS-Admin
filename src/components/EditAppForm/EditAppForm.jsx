import { Formik, ErrorMessage, Field } from 'formik';
import * as Yup from 'yup';
import { Form, Button } from 'react-bootstrap';
import { toast } from 'react-toastify';

const AppSchema = Yup.object().shape({
  name: Yup.string()
    .min(5, 'Too Short!')
    .max(50, 'Too Long!')
    .required('Name is required'),
});

const EditAppForm = ({ initialValues, onEdit }) => {
  const onSubmit = (values) => {
    onEdit(values);

    toast('Application info updated successfully');
  };

  return (
    <Formik
      initialValues={initialValues}
      validationSchema={AppSchema}
      onSubmit={onSubmit}
    >
      {({ errors, touched, handleSubmit }) => (
        <Form>
          <Form.Group>
            <Form.Label>Name</Form.Label>
            <Form.Control
              as={Field}
              name='name'
              type='text'
              placeholder='Enter name of app'
              isInvalid={errors.name && touched.name}
            />
            {errors.name && touched.name ? (
              <Form.Control.Feedback type='invalid'>
                <ErrorMessage name='name' />
              </Form.Control.Feedback>
            ) : null}
          </Form.Group>
          <Form.Group>
            <Form.Label>App ID</Form.Label>
            <Form.Control as={Field} name='id' type='text' readOnly />
          </Form.Group>

          <Button type='button' variant={'info'} onClick={handleSubmit}>
            Save Changes
          </Button>
        </Form>
      )}
    </Formik>
  );
};

export default EditAppForm;
