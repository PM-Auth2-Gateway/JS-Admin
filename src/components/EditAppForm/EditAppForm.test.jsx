import React, { useState as useStateMock } from 'react';
import { shallow } from 'enzyme';

import EditAppForm from './EditAppForm';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

describe('EditAppForm Component', () => {
  const setState = jest.fn();

  beforeEach(() => {
    useStateMock.mockImplementation((init) => [init, setState]);
  });

  it('should render EditAppForm Component', function () {
    const component = shallow(<EditAppForm />);

    expect(component).toMatchSnapshot();
  });
});
