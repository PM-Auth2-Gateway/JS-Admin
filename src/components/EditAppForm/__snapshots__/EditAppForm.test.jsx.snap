// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAppForm Component should render EditAppForm Component 1`] = `
<Formik
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_excludedEdges": Array [],
      "_nodes": Array [
        "name",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "name": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "max": true,
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <Component />
</Formik>
`;
